torch/optim/adam.py

```diff
@@ -105,7 +105,7 @@ def _init_group(
                     state['step'] = (
                         torch.zeros((), dtype=torch.float, device=p.device)
                         if group['capturable'] or group['fused']
-                        else torch.tensor(0.)
+                        else torch.tensor(0., device=p.device)
```

Comment:
I think there were some tricky considerations for this, because `torch.zeros((), dtype=torch.float, device=p.device)` is actually equivalent to `torch.tensor(0., device=p.device)`, so with this fix the if-else-switch is vacuous... 

But I don't know how it should be, other people would know :) Also, regardless of which is right, it would be nicer to use the same syntax in both branches: torch.zeros or torch.tensor